test/lolex-test.js

```diff
@@ -114,6 +114,93 @@ describe("issue sinon#1852", function () {
     });
 });
 
+describe("issue #207 - nanosecond round-off errors on high-res timer", function () {
+    var clock;
+
+    afterEach(function () {
+        clock.uninstall();
+    });
+
+    if (hrtimePresent) {
+        it("should not round off nanosecond arithmetic on hrtime - case 1", function () {
+            clock = lolex.install();
+
+            clock.tick(1022.7791);
+
+            var nanos = clock.hrtime([0, 2 * 1e7])[1];
+            assert.equals(nanos, 2779100);
+        });
+
+        it("should not round off nanosecond arithmetic on hrtime - case 2", function () {
+            clock = lolex.install({
+                now: new Date("2018-09-12T08:58:33.742000000Z").getTime(),
+                toFake: ["hrtime"]
+            });
+            var start = clock.hrtime();
+            clock.tick(123.493);
+
+            var nanos = clock.hrtime(start)[1];
+            assert.equals(nanos, 123493000);
+        });
+
+        it("should truncate sub-nanosecond ticks", function () {
+            clock = lolex.install();
+            clock.tick(0.123456789);
+
+            var nanos = clock.hrtime()[1];
+            assert.equals(nanos, 123456);
+        });
+    }
+
+    it("should always set 'now' to an integer value when ticking with sub-millisecond precision", function () {
+        clock = lolex.install();
+        clock.tick(2.993);
+
+        assert.equals(clock.now, 2);
+    });
+
+    it("should adjust adjust the 'now' value when the nano-remainder overflows", function () {
+        clock = lolex.install();
+        clock.tick(0.993);
+        clock.tick(.5);
+
+        assert.equals(clock.now, 1);
+    });
+
+    it("should floor negative now values", function () {
+        clock = lolex.install({now: -1.2});
+
+        assert.equals(clock.now, -2);
+    });
+
+    it("should floor start times", function () {
+        clock = lolex.install({now: 1.2});
+        assert.equals(clock.now, 1);
+    });
+
+    it("should floor negative start times", function () {
+        clock = lolex.install({now: -1.2});
```

Comment:
does it make sense with negative start times?

Reply:
Yes, that just means time before 1970 :)